### ISSUE TITLE
[Outreachy Task Submission] Survey Select All button on data view not working properly.

### ISSUE THREAD

**Issue by kivenalaric** [ushahidi/platform#4759]
**Describe the bug**
When the **select all** button is clicked, it doesnt actually select all as it should whereas the clear list works fine.

**Where was the bug observed**
http://localhost:4200/feed

### To Reproduce
1. Go to 'Data view'.
2. select surveys.
3. Click on clear all
4. then click on select all
5. Doesnt select all

### Expected behavior
A click on the select all button should mark all survey checkboxes as checked.

**Screenshots**
<img width="1680" alt="Screenshot 2024-03-07 at 09 07 42" src="https://github.com/ushahidi/platform/assets/89359876/0962c360-8584-4765-adfd-00ce6190ec62">

[57833ac9-f11d-4e71-be53-12d3a71eaec7.webm](https://github.com/ushahidi/platform/assets/89359876/21db4254-801d-4441-a7a8-fbe490e53366)

**URL / Environment where this happened**
http://localhost:4200/feed

**Desktop Hardware Details (please complete the following information):**
 - Mac OS
 - Browser: Chrome
 - Version: Sanoma

**Comment by mhdamaan79** [comment]
Hi @Angamanga, eager to work on this. Please assign it to me. All steps and setup are done.

**Comment by kivenalaric** [comment]
Hi @Angamanga, would like to work on this. Please assign it to me. Setup already done

**Comment by Angamanga** [comment]
@kivenalaric Thank you for raising this issue. I have been aware of this but haven't had time to add an issue for it so I am really thankful for someone else raised it :) Please go ahead! @mhdamaan79 Since @kivenalaric raised the issue I am giving him the priority here.

**Comment by Angamanga** [comment]
@kivenalaric If you look at how it is done in the same view but for sources, where the button is actually working, you may get some hints.

**Comment by mhdamaan79** [comment]
@Angamanga, If @kivenalaric  is unable to work on this issue, I'm available and eager to contribute. Please consider assigning it to me.

**Comment by kivenalaric** [comment]
@Angamanga i created a pr yesterday and tagged you there
